src/library/scala/sys/process/ProcessImpl.scala

```diff
@@ -147,6 +147,7 @@ private[process] trait ProcessImpl {
           throw err
         }
       runInterruptible {
+        source.join()
         val exit1 = first.exitValue()
```

Comment:
Hi @som-snytt , Yes, Since the `Spawn` start a new `ThreadProcess` to run the `PipedProcess`, but `I/O` is also async running. so **uncomment** `thread.join()` will not fix this. we need to wait the `I/O` by `source.join()`  in the new  `ThreadProcess`.